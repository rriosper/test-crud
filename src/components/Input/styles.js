import { css } from "@xstyled/styled-components";
import { prop } from "ramda";
import { whenAlways } from "ramdu";

export const fancyStyle = whenAlways(
  prop("fancy"),
  css(`
  opacity: 0.5;

  : hover {
    opacity: 1;
  }
`)
);

export const asTypeStyle = ({ as: asType }) => {
  return asType && css({ minHeight: "xl" });
};
