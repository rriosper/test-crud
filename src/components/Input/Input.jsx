import styled from "@xstyled/styled-components";
import {
  margin,
  marginBottom,
  marginLeft,
  marginRight,
  marginTop,
  width,
} from "@xstyled/system";
import PropTypes from "prop-types";
import { Input as RInput } from "reakit";

import { createCleanComponent } from "#utils";

import { asTypeStyle, fancyStyle } from "./styles";

const CleanInput = createCleanComponent(RInput, ["fancy", "width"]);

const Input = styled(CleanInput)`
  ${asTypeStyle}
  ${fancyStyle}

  ${width}
  ${margin}
  ${marginBottom}
  ${marginLeft}
  ${marginRight}
  ${marginTop}

  outline: none;

  color: black;
  padding: sm;
  font-size: md;
  border-color: border;
  transition: default;
  border-radius: default;
  border: default;
  height: lg;
  
`;

Input.defaultProps = {
  fancy: false,
  as: undefined,
  width: "100%",
};

Input.propTypes = {
  fancy: PropTypes.bool,
  as: PropTypes.oneOf(["textarea"]),
};

export default Input;
