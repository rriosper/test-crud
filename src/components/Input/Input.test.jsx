import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";

import Input from "./Input";

afterEach(cleanup);

describe("Input", () => {
  test("Default type", () => {
    let value = "";
    const onChange = jest.fn(({ target: { value: newValue } }) => {
      value = newValue;
    });
    const { getByPlaceholderText, asFragment } = render(
      <Input placeholder="Placehoder text" onChange={onChange} value={value} />
    );

    expect(onChange).toHaveBeenCalledTimes(0);
    fireEvent.change(getByPlaceholderText("Placehoder text"), {
      target: {
        value: "Updated text!",
      },
    });
    expect(onChange).toHaveBeenCalledTimes(1);

    expect(value).toBe("Updated text!");
    expect(asFragment()).toMatchSnapshot();
  });
});
