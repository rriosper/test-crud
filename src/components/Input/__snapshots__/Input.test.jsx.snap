// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Default type 1`] = `
<DocumentFragment>
  .c0 {
  placeholder: Placehoder text;
  width: 100%;
  width: 100%;
  outline: none;
  color: black;
  padding: sm;
  font-size: md;
  border-color: border;
  -webkit-transition: default;
  transition: default;
  border-radius: default;
  border: default;
  height: lg;
}

.c0 on-change:function mockConstructor(_a) {
  return: fn.apply(this,arguments);
}

<input
    class="c0"
    placeholder="Placehoder text"
    value=""
  />
</DocumentFragment>
`;
